test/lib/Elastica/Test/ResultSetTest.php

```diff
@@ -58,6 +58,35 @@ public function testArrayAccess()
 
     /**
      * @group functional
+     */
+    public function testDocumentsAccess()
+    {
+        $index = $this->_createIndex();
+        $type = $index->getType('test');
+
+        $type->addDocuments(array(
+            new Document(1, array('name' => 'elastica search')),
+            new Document(2, array('name' => 'elastica library')),
+            new Document(3, array('name' => 'elastica test')),
+        ));
+        $index->refresh();
+
+        $resultSet = $type->search('elastica search');
+
+        $this->assertInstanceOf('Elastica\ResultSet', $resultSet);
+        
+        $documents = $resultSet->getDocuments();
+
+        $this->assertInternalType('array', $documents);
+        $this->assertEquals(3, count($documents));        
+        $this->assertInstanceOf('Elastica\Document', $documents[0]);
+        $this->assertInstanceOf('Elastica\Document', $documents[1]);
+        $this->assertInstanceOf('Elastica\Document', $documents[2]);
+        $this->assertFalse(isset($documents[3]));
```

Comment:
Make no sense, since there is assertEquals - 3 an than all three you check in next lines.

Better here to test document content, like name => 'elastica search' is really in document.